src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,114 @@
+                    throw new MissingTaskIndexException();
+                }
+                return new DeleteCommand(Integer.valueOf(wordsArray[1]) - 1);
+            } else if (wordsArray[0].equals("find")) {
+                if (wordsArray.length != 2) {
+                    throw new MissingKeywordException();
+                }
+                return new FindCommand(wordsArray[1]);
+            } else if (wordsArray[0].equals("todo") || wordsArray[0].equals("event")
+                    || wordsArray[0].equals("deadline")) {
+                // Declaring the variables required for creating a Task instance
+                String taskType;
+                String task;
+                LocalDate date;
+
+                // Checks if a task description is provided by the user
+                taskType = wordsArray[0];
+                int indexOfSpaceChar = command.indexOf(" ");
+                if (indexOfSpaceChar == -1) {
+                    throw new EmptyTaskDescriptionException(taskType);
+                }
+                String taskDescription = command.substring(indexOfSpaceChar + 1);
+                if (taskDescription.trim().isEmpty()) {
+                    throw new EmptyTaskDescriptionException(taskType);
+                }
+
+                // Parsing the user input based on the task type
+                if (taskType.equals("deadline")) {
+                    if (!taskDescription.contains(" /by ")) {
+                        throw new TimeNotSpecifiedException(" /by ");
+                    }
+                    String[] descriptionArr = taskDescription.split(" /by ");
+                    task = descriptionArr[0];
+                    String deadline = descriptionArr[1];
+                    if (deadline.trim().isEmpty()) {
+                        throw new TimeNotSpecifiedException(" /by ");
+                    }
+                    date = LocalDate.parse(deadline);
+                    return new AddCommand(taskType, task, date);
+                } else if (taskType.equals("event")) {
+                    if (!taskDescription.contains(" /at ")) {
+                        throw new TimeNotSpecifiedException(" /at ");
+                    }
+                    String[] descriptionArr = taskDescription.split(" /at ");
+                    task = descriptionArr[0];
+                    String timeFrame = descriptionArr[1];
+                    if (timeFrame.trim().isEmpty()) {
+                        throw new TimeNotSpecifiedException(" /at ");
+                    }
+                    date = LocalDate.parse(timeFrame);
+                    return new AddCommand(taskType, task, date);
+                } else {
+                    task = taskDescription;
+                    return new AddCommand(taskType, task);
+                }
+            } else {
+                throw new UnrecognisedCommandException();
+            }
```

Comment:
The logic in this block looks rather convoluted. Perhaps the use of switch statements can help?

Reply:
Good point, I will make changes to this block of code! Thanks for the catch, and also for the suggestion! 😄